### PR TITLE
Add tests and refactor

### PR DESCRIPTION
**Pull request by digitalsadhu**
## JIRA Issue
[COREWEB-196](https://jira.finn.no/browse/COREWEB-196)

## Status
**IN DEVELOPMENT**

## Description
* Adds tests.
* Adds assertions around module input.
* Refactors module into browserify plugins.
* Removes minification.
* Removes auto deduping by browserify.
* Refactors into small modules.

**BREAKING CHANGE**: minify option no longer available
**BREAKING CHANGE**: feeds no longer deduped by default

**Why??:**
1. Changes to @asset-pipe/js-reader require that feeds no longer do any deduping as this now happens in the reader and doing so before hand causes all kinds of mischief.
2. Codebase and repository cleanup.
3. Increase test coverage.

## Todos
- [x] Tests - additional more advanced integration tests still to add
- [x] Documentation - document api changes

## Deploy Notes
- [ ] Publish
- [ ] update @asset-pipe/client with new version of this module
- [ ] update (and expand?) integration tests
- [ ] update@finn-no/podlet module
- [ ] update all podlets using@finn-no/podlet module and re-deploy

## Related PRs
* None

### CODE REVIEW

File: test/helpers/get-stream.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (SimenB) [inline test/helpers/get-stream.js]: copy paste :(

File: lib/writer.js [hunk 2]
```diff
@@ -1,64 +1,29 @@
...
+        assert(
+            typeof options === 'object',
+            `Expected optional 'options' argument to ${name} constructor to be an object. Instead got '${typeof options}'`
+        );
+        assert(
+            typeof bundle === 'boolean',
+            `Expected optional 'bundle' argument to ${name} constructor to be a boolean. Instead got '${typeof bundle}'`
+        );
+
+        const opts = { dedupe: false, ...options };
```
Comment (SimenB) [inline lib/writer.js]: should we allow users to overwrite this?

Comment (SimenB) [reply lib/writer.js]: "this" being `dedupe`

Comment (digitalsadhu) [reply lib/writer.js]: probably not eh?

**Pull request merged**